datastore/datastore.go

```diff
@@ -37,23 +50,35 @@ type KV interface {
 const (
 	// NetworkKeyPrefix is the prefix for network key in the kv store
 	NetworkKeyPrefix = "network"
+	// EndpointKeyPrefix is the prefix for endpoint key in the kv store
+	EndpointKeyPrefix = "endpoint"
 )
 
+var rootChain = []string{"docker", "libnetwork"}
+
 //Key provides convenient method to create a Key
 func Key(key ...string) string {
-	keychain := []string{"docker", "libnetwork"}
-	keychain = append(keychain, key...)
+	keychain := append(rootChain, key...)
 	str := strings.Join(keychain, "/")
 	return str + "/"
 }
 
+//UnKey provides convenient method to unpack the key to compliment the Key function
+func UnKey(key string) ([]string, error) {
+	chain := strings.Split(key, "/")
+	if len(chain) <= len(rootChain) || !reflect.DeepEqual(chain[0:len(rootChain)], rootChain) {
+		return nil, types.BadRequestErrorf("Invalid Key : %s", key)
+	}
+	return chain[len(rootChain):], nil
```

Comment:
You are right. this needs `strings.Trim` for `/`
But fortunately, this didn't impact us as the keys were already trimmed off these. But we should not be relying on that. Will add the `strings.Trim` in the next update.